sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/test/TestTableProvider.java

```diff
@@ -211,9 +211,8 @@ public BeamTableStatistics getTableStatistics(PipelineOptions options) {
         }
       }
 
-      // When project push-down is supported.
-      if ((options == PushDownOptions.PROJECT || options == PushDownOptions.BOTH)
-          && !fieldNames.isEmpty()) {
+      // When project push-down is supported or field reordering is needed.
```

Comment:
This is interesting. I wouldn't expect an IO to support field reordering unless it also supports push-down. Can you explain what is going on here?

Reply:
The problem I was trying to fix by modifying that `if` statement is a scenario when an IO only supports Filter push-down and predicate can be completely pushed-down to IO layer, all fields are selected, so there is no need to preserve the `Calc` to perform project, but selected fields are not in the random order.
In the scenario described above an IO does need to reorder fields (either with a `Select` or by not dropping a `Calc`).

I agree that the current `if` statement is incorrect and it should look more like what it used to, but with an additional check. Not quite sure how to check that the `IORel` is not followed by a `CalcRel` and that the fields are not selected in the order they are present in the original schema, but I'm working on it.

Reply:
Decided to not drop the `Calc` when fields are selected in a different order for now.